packages/core/src/modules/connections/repository/ConnectionRecord.ts

```diff
@@ -26,10 +26,11 @@ export interface ConnectionRecordProps {
   threadId?: string
   tags?: CustomConnectionTags
   imageUrl?: string
-  multiUseInvitation: boolean
+  multiUseInvitation?: boolean
   mediatorId?: string
   errorMessage?: string
   protocol?: HandshakeProtocol
+  outOfBandId?: string
```

Comment:
Thinking whether it may be better to give the oob record an optional connectionId? A connection can have multiple oob records associated with it (due to connection reuse), while each oob record can have at most 1 connection.

Reply:
Hmm. Isn't it actually vice-versa? I can create a multi-use OOB record. Therefore I'll have more connections related to one OOB. If I reuse a connection, it doesn't create a new one.

Reply:
Hmm, maybe you're right.

But in the following scenario it's the other way around:
1. I create an oob invitation with handshake protocols
2. Other party receices the invitation, a connection is made.
3. I make another oob invitation with handshake.
The other party receives it and sends an connection reuse.
4. I now have 2 oob record that are associated witht the same connection.


With multi use invitations there's indeed the possibility of multiple connections having the same oob record.....